### ISSUE TITLE
SC.DragHandle mixin for exampleViews - turn any child view into a live-togglable drag handle

### ISSUE THREAD

**Issue by dcporter** [sproutcore/sproutcore#824]
This idea came out of a previous discussion (https://github.com/sproutcore/sproutcore/issues/714) about the inability to rearrange list items on an iPad. (Reason being: the touch-and-drag event sequence is indistinguishable from the touch-and-scroll sequence.)

My original idea was to solve the problem at hand with a slightly different event sequence (touch-hold-then-drag), but @geoffreyd rightly pointed out that this is different than any other implementation anywhere, and would have the discoverability of a rutabaga. He noted that other implementations of rearranging-lists-by-touch involve the exposure of an "edit mode", or the use of a dedicated drag handle.

I took a step back and decided that the problem could be reduced to how to easily enable drag handles. If any arbitrary child of an example view could be turned into a drag handle - and if that behavior could be turned on and off via your standard KVO options - then developers could choose how, and when, their users could rearrange a list.

I thought a mixin would make most sense, since the ability to become a drag handle needs to be able to be mixed into any arbitrary view. Something like this:

``` javascript
SC.DragHandle = {
  isDragHandle: YES,
  dragIsEnabled: YES,
  // one of several event handlers that would be required.
  mouseDown: function(evt) {
    // mark event as involving a drag handle; this will trigger drag behavior from any upstream view that supports it
    if (this.get('dragIsEnabled')) evt.isDragHandle = YES;
    // keep propagating event
    return NO;
  }
}
```

The collection view would then trigger drag behavior any time the event came in with isDragHandle set to YES. (For backward-compatibility, "canReorderContent" could still be obeyed too, but it would be deprecated.) This behavior would be triggered (and would override touch scrolling for affected touches) regardless of other considerations, so the developer would be in charge of turning it on or off in different scenarios, but this would be easy:
- If a developer wanted to mimic the current behavior, she would mix the mixin to the root exampleView, and set dragIsEnabled to a calculated property which returned YES when on a mousey device and NO when on a touchy device. (This could remain the default behavior.)
- If a developer wanted to expose an edit mode, she could include a drag handle view with its isVisible tied to a controller somewhere that kept track of whether the list was in edit mode or not.
- Alternately, a draggy-looking handle could be permanently visible on all devices, with all drag operations going through this element.
- It would be trivial to implement a hybrid solution, where the exampleView root view is an active drag handles on mouse devices and not on touch devices, with a smaller drag handle set to be visible on touch devices.
- If developers are implementing advanced renderers in their exampleViews for better performance (see for example SC.ListItemView), the developer could override the event handlers to set evt.isDragHandle to YES depending on classnames of evt.srcElement.

Long story short, it's a simple, flexible solution, with sane defaults and the ability to implement a variety of alternative solutions.

Thoughts?
